### PR TITLE
boards: rename nucleo144-f207 to nucleo-f207zg

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

This PR renames the board "nucleo144-f207" to its marketing name "nucleo-f207zg".

### Issues/PRs references

#8096

### CODE REVIEW

**Comment by aabadie** [comment]
Thanks for introducing this. I also wanted to do that change after #8065 was merged. There are quite a few to change...

**Comment by kaspar030** [comment]
> There are quite a few to change...

I have a script that I'm testing with this PR... ;)

**Comment by kaspar030** [comment]
@aabadie would you mind ACKing? I think it is a rather minor change, and there are ~15 more of it...

**Comment by aabadie** [comment]
> would you mind ACKing?

Yes but I'd like to have #8432 merged before. It will be 'simpler' to rebase this simple one after it's merged.

> I think it is a rather minor change, and there are ~15 more of it...

I think all nucleo could be renamed with a single PRs, Or maybe one PR per nucleo family (32, 64, 144). What do you think ?

**Comment by kaspar030** [comment]
- rebased

**Comment by kaspar030** [comment]
> I think all nucleo could be renamed with a single PRs, Or maybe one PR per nucleo family (32, 64, 144). What do you think ?

I will put the others in one PR, if they are simple search&replace like this one.

File: boards/nucleo-f207zg/include/periph_conf.h [hunk 1]
```diff
@@ -7,13 +7,13 @@
  */
 
 /**
- * @defgroup    boards_nucleo144-f207 STM32 Nucleo144-F207
+ * @defgroup    boards_nucleo-f207zg STM32 Nucleo-F207ZG
  * @ingroup     boards_common_nucleo144
- * @brief       Support for the STM32 Nucleo144-F207
+ * @brief       Support for the STM32 Nucleo144-F207ZG
```
Comment (aabadie) [inline boards/nucleo-f207zg/include/periph_conf.h]: Should it be Nucleo-F207ZG ?

Comment (kaspar030) [reply boards/nucleo-f207zg/include/periph_conf.h]: totally. fixed.

**Pull request merged**